File: core/src/main/java/inetsoft/web/viewsheet/event/OpenPreviewViewsheetEvent.java
/*
 * This file is part of StyleBI.
 * Copyright (C) 2024  InetSoft Technology
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */
package inetsoft.web.viewsheet.event;

/**
 * Class that encapsulates the parameters for opening a preview viewsheet.
 *
 * @since 12.3
 */
public class OpenPreviewViewsheetEvent extends OpenViewsheetEvent {
   /**
    * Gets the name of the current layout ot preview, if available.
    *
    * @return the name of the layout.
    */
   public String getLayoutName() {
      return layoutName;
   }

   /**
    * Sets the name of the current layout ot preview, if available.
    *
    * @param layoutName the name of the layout.
    */
   public void setLayoutName(String layoutName) {
      this.layoutName = layoutName;
   }

   @Override
   public String toString() {
      return "OpenPreviewViewsheetEvent{" +
         "layoutName='" + layoutName + '\'' +
         '}';
   }

   private String layoutName;
}

File: web/projects/portal/src/app/vsobjects/event/open-preview-viewsheet-event.ts
/*
 * inetsoft-web - StyleBI is a business intelligence web application.
 * Copyright © 2024 InetSoft Technology (info@inetsoft.com)
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affrero General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
import { OpenViewsheetEvent } from "./open-viewsheet-event";

/**
 * Event used to open a preview viewsheet.
 */
export class OpenPreviewViewsheetEvent extends OpenViewsheetEvent {
   /**
    * The name of the current layout if available.
    */
   public layoutName: string;

   /**
    * Creates a new instance of <tt>OpenPreviewViewsheetEvent</tt>.
    *
    * @param entryId   the asset entry identifier of the viewsheet.
    * @param width     the viewport width of the browser.
    * @param height    the viewport height of the browser.
    * @param mobile    the flag that indicates if the client is a mobile device.
    * @param userAgent the user agent string of the client browser.
    */
   constructor(entryId: string, width: number, height: number, mobile: boolean,
               userAgent: string)
   {
      super(entryId, width, height, mobile, userAgent);
   }
}